### PR TITLE
Move timetrackings/start to thunk

### PR DESCRIPTION
**Pull request by benj56**
Also interesting: https://redux-toolkit.js.org/api/createAsyncThunk#handling-thunk-errors

### CODE REVIEW

File: src/entities/timeTrackings/timeTrackingsSlice.ts [hunk 1]
```diff
@@ -8,9 +8,15 @@ import {
...
 
+import { timeTacApi } from '../../dependencies';
 import { timeTrackingsAdapter } from './timeTrackingsAdapter';
 
-export const timeTrackingsSlice = createSlice({
+const start = createAsyncThunk(
+  'timeTrackings/start',
+  async (timeTracking: StartTimeTrackingData) => await timeTacApi.timeTrackings.start(timeTracking)
+);
+
```
Comment (benj56) [inline src/entities/timeTrackings/timeTrackingsSlice.ts:18]: Generates 3 actions: `timeTrackings/start/pending`, `timeTrackings/start/fulfilled`, `timeTrackings/start/rejected`.
See https://redux-toolkit.js.org/api/createAsyncThunk#parameters

**Review by sergej-tschak — APPROVED** [review]

File: src/entities/timeTrackings/timeTrackingsSlice.ts [hunk 2]
```diff
@@ -43,7 +49,18 @@ export const timeTrackingsSlice = createSlice({
     },
     requestCurrentRunning: (_state, _action: PayloadAction) => undefined,
     requestPreviousRunning: (_state, _action: PayloadAction) => undefined,
-    requestStart: (_state, _action: PayloadAction<StartTimeTrackingData>) => undefined,
     requestStop: (_state, _action: PayloadAction<StopTimeTrackingData>) => undefined,
   },
+  extraReducers: (builder) => {
+    builder.addCase(start.fulfilled, (state, action) => {
```
Comment (whosshell) [inline src/entities/timeTrackings/timeTrackingsSlice.ts:55]: Can't you just write:
```
[start.fullfilled]: (state, action) => {
        ...action.payload.results,
        // TODO: CL Resource response types are still wrong, affected must be nullable, so we need the disable below
        // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
        ...((action.payload.affected.timeTrackings ?? []) as TimeTracking[]),
}
```

Comment (whosshell) [reply src/entities/timeTrackings/timeTrackingsSlice.ts:55]: As discussed for stricter type safety preferred way of doing it, is with [builder.addCase](https://redux-toolkit.js.org/usage/usage-with-typescript#building-type-safe-reducer-argument-objects)

**Comment by whosshell** [comment]
Before merging, please make sure that this works with web.

**Pull request merged**